pkg/es/config/config.go

```diff
@@ -329,12 +347,17 @@ func (tlsConfig *TLSConfig) loadPrivateKey() (*tls.Certificate, error) {
 
 // TokenAuthTransport
 type tokenAuthTransport struct {
-	token   string
-	wrapped *http.Transport
+	token                string
+	allowOverrideFromCtx bool
+	wrapped              *http.Transport
 }
 
 func (tr *tokenAuthTransport) RoundTrip(r *http.Request) (*http.Response, error) {
-	r.Header.Set("Authorization", "Bearer "+tr.token)
+	token := tr.token
+	if tr.allowOverrideFromCtx {
+		token, _ = spanstore.GetBearerToken(r.Context())
```

Comment:
Would be better to return the token into a local var, check a value has been found, before overwriting the `token`. That way - if a request is sent in without a bearer token, it will still revert to the supplied `tr.token`.

Reply:
well, I wasn't sure if we wanted to do that, but ok I'll do the change